Formula/termshark.rb

```diff
@@ -0,0 +1,24 @@
+class Termshark < Formula
+  desc "Terminal UI for tshark, inspired by Wireshark"
+  homepage "https://termshark.io"
+  url "https://github.com/gcla/termshark/archive/v1.0.0.tar.gz"
+  sha256 "669bba0e8dd7df54ade6321a5c7d2ec20563ffd777f7b3b0394a11f88da64698"
+
+  depends_on "go@1.11" => :build
+
+  def install
+    # Don't set GOPATH because we want to build using go modules to
+    # ensure our dependencies are the ones specified in go.mod
+    ENV["GO111MODULE"] = "on"
+    mkdir_p buildpath
+    ln_sf buildpath, buildpath/"termshark"
+
+    cd "termshark" do
+      system "go", "build", "-o", bin/"termshark", "-ldflags", "-X github.com/gcla/termshark.Version=#{version}", "cmd/termshark/termshark.go"
+    end
+  end
+
+  test do
+    assert_match "termshark v1.0.0", shell_output("#{bin}/termshark -v")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thanks for the feedback. I'll rework this formula by adding additional tests and make a new PR.

Reply:
No need to make a new PR. You can simply force push to this one. In fact, that's preferable!

Reply:
Or even just push new commits to the branch this PR is based on, the PR will update. We can squash your commits for you when we merge.

Reply:
Also, it looks like something is wrong with the install. This simple test is failing with:
```
==> /usr/local/Cellar/termshark/1.0.0/bin/termshark -v
Error looking up tshark binary: exec: "tshark": executable file not found in $PATH
Error: termshark: failed
An exception occurred within a child process:
  Test::Unit::AssertionFailedError: <0> expected but was
<1>.
```

Reply:
Yes you're right, my formula is incomplete 😪 And it shows why I should've added proper tests like you suggested. Termshark needs tshark at runtime and a proper test would've caught that. I've added Wireshark as a dependency but am running into some issues building it. More soon!

Reply:
Hi - I've pushed an update to the formula that includes some better thought-out tests.